specs/HubProtocol.md

```diff
@@ -348,8 +363,160 @@ World;0:;
 
 Note that the final frame still ends with the `;` terminator, and that since the body may contain `;`, newlines, etc., the length is specified in order to know exactly where the body ends.
 
+## MessagePack (MsgPack) encoding
+
+In the MsgPack Encoding of the SignalR Protocol, each Message is represented as a single MsgPack array containing items that correspond to properties of the given hub protocol message. The array items may be primitive values, arrays (e.g. method arguments) or objects (e.g. argument value). The first item in the array is the message type.
```

Comment:
1. Versioning is a different story (and not part of this change). The way I think it should work is that the Negotiation message contains the version and then we could have version-specific implementations of HubProtocol. This way a new version of protocol is completely independent of the old version.

2. We could but we will give up readability. Also requires a little bit more code on our side (especially in TS client).